doc/shared/sundomeigest/SUNDomEigEst_API.rst

```diff
@@ -0,0 +1,622 @@
+..
+   Mustafa Aggul @ SMU
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2025, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNDomEigEst.API:
+
+The SUNDomEigEstimator API
+=============================
+
+The SUNDomEigEst API defines several dominant eigenvalue estimation operations that enable
+SUNDIALS packages to utilize this API.  These functions can be divided into three categories.
+The first are the core dominant eigenvalue estimation functions.  The second consist of "set"
+routines to supply the dominant eigenvalue estimator with functions provided by the SUNDIALS
+packages and to modify estimator parameters.  The final group consists of "get" routines for
+retrieving dominant eigenvalue estimation statistics.  All of these functions are defined in
+the header file ``sundials/sundials_domeigestimator.h``.
+
+
+.. _SUNDomEigEst.CoreFn:
+
+SUNDomEigEstimator core functions
+-----------------------------------------------------
+
+The core dominant eigenvalue estimator functions consist of several **required**
+functions: :c:func:`SUNDomEigEst_NewEmpty` creates a new empty ``SUNDomEigEstimator`` 
+object, :c:func:`SUNDomEigEst_SetATimes` provides a :c:type:`SUNATimesFn` function pointer,
```

Comment:
I wouldn't label `SUNDomEigEst_NewEmpty` as a "required" function, since it is a utility routine provided to implementors (who never need to write this themselves).  I recommend something more like:
```suggestion
The SUNDomEigEstimator base class provides two **utility** routines for implementors, :c:func:`SUNDomEigEst_NewEmpty` and :c:func:`SUNDomEigEst_FreeEmpty`.

Implementations of SUNDomEigEstimators must include a set of **required** functions: `
:c:func:`SUNDomEigEst_SetATimes` provides a :c:type:`SUNATimesFn` function pointer,
```